modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/fds_view/fields/modal_content/AddFieldsModalContent.tsx

```diff
@@ -5,10 +5,15 @@
 
 import ClayButton from '@clayui/button';
 import {TreeView} from '@clayui/core';
-import {ClayCheckbox} from '@clayui/form';
+import {ClayCheckbox, ClayInput} from '@clayui/form';
+import ClayIcon from '@clayui/icon';
 import ClayLoadingIndicator from '@clayui/loading-indicator';
+import ClayManagementToolbar, {
+	ClayResultsBar,
+} from '@clayui/management-toolbar';
 import ClayModal from '@clayui/modal';
-import {fetch} from 'frontend-js-web';
+import {useDebounce} from '@clayui/shared';
```

Comment:
I'm not sure, you are right that it can be simpler. But as a user, I expect the filters to be applied when I have finished typing and not every change. The debounce version feels better than the simple version.

How the frontend filtering is implemented along the DSM?

Reply:
We implemented all filtering without debouncing. See for example in:
- [FDSEntries](https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/FDSEntries.tsx#L98-L110)
- [OrderableTable](https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/components/OrderableTable.tsx#L311-L331) 
- [ActionsForm](https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/fds_view/actions/ActionForm.tsx#L347-L363)
- [Fields](https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/fds_view/fields/Fields.tsx#L146-L161)

But, you may have a point with customers' expectation. We don't _need_ to implement it that way, but we may _want_ to. @ugeortiz What do you think? It is better to wait with search until user stops typing? To me, it seems like immediate results are better, but that might not be ergonomic for a typical user.

Reply:
Ty! I'll take a look.

Maybe I can remove the debounce for now, and apply it in the future